microsoft/testsuites/lis/lissuite.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CentOs, Redhat
+from lisa.sut_orchestrator.azure.tools import LisDriver
+from lisa.testsuite import simple_requirement
+from lisa.tools import Wget
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="lis",
+    category="functional",
+    description="""
+    ##################################
+    """,
+    requirement=simple_requirement(
+        supported_os=[
+            CentOs,
+            Redhat,
+        ]
+    ),
+)
+class Lis(TestSuite):  # noqa
+    @TestCaseMetadata(
+        description="""
+            #################
+        """,
+        priority=1,
+    )
+    def verify_lis_driver_version(self, node: Node, log: Logger) -> None:
+        wget_tool = node.tools[Wget]
+        node.tools[LisDriver]
+
+        # Checking for RMP
+        rpm_status = node.execute(
+            'rpm -qa | grep "kmod-microsoft-hyper-v" && rpm -qa '
+            '| grep "microsoft-hyper-v"'
+        ).exit_code
+        if rpm_status != 0:
+            raise SkippedException("No LIS RPM's are detected. Skipping test.")
+
+        # Capturing LIS version from the node
+        version = node.execute(
+            """modinfo "hv_vmbus" | grep version: | head -1 | awk '{print $2}'""",
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+        ).stdout
+        print("modinfo: ", version)
+
+        for i in range(5, 8):
+            node.execute("rm -rf hv_compat.h")
+            wget_tool.get(
+                "https://raw.githubusercontent.com/LIS/lis-next/"
+                f"{version}/hv-rhel{i}.x/hv/include/linux/hv_compat.h",
+                filename="hv_compat.h",
+                file_path="./",
+            )
+
+            # Capturing the LIS version from the source code
+            source_version = node.execute(
+                """grep 'define HV_DRV_VERSION' hv_compat.h|cut -d '"' -f 2""",
```

Comment:
if the file content is not long (no more than 200 lines). cat the content to Python, and use the regexp to parse it. If it's too long, use grep to filter out, and also use python to parse the line.

Reply:
It's more than 200 lines

Reply:
So, use the grep to get this line printed only, and parse it in Python.

Reply:
you can add a new method in `Cat` called `read_with_filter`. It integrates the cat and grep in one line.

Reply:
Do you think it would be better if we just created a grep tool?

Reply:
grep is not recommended in LISA. It hides the raw information, and it's difficult to debug from the raw output. It's used in limited area, like long output or extremely high performance is needed.